protocol-models/src/main/resources/airbyte_protocol/airbyte_protocol.yaml

```diff
@@ -584,6 +584,60 @@ definitions:
       - overwrite
       #- upsert_dedup # TODO chris: SCD Type 1 can be implemented later
       - append_dedup # SCD Type 1 & 2
+      - insert
+      - update
+      - upsert
+      - soft_delete
+  DiscoveredDestination:
+    type: object
+    additionalProperties: true
+    description: Return value for a discover command on a catalog. This should indicate which operations are supported for which objects.
+    required:
+      - stream_name
+      - operations
+    properties:
+      stream_name:
+        description: The name of the stream
+        type: string
+      operations:
+        description: The operations supported for this stream
+        type: array
+        items:
+          "$ref": "#/definitions/DestinationOperation"
+  DestinationOperation:
+    type: object
+    additionalProperties: true
+    description: An operation supported on a destination object including fields that are involved in this operation.
+    required:
+      - mode
+      - mandatory_fields
+      - record_matcher
+    properties:
+      additional_fields:
+        description: Whether the operation allows fields that are not defined as part of the mandatory or available fields.
+        type: boolean
+        default: false
+      available_fields:
+        description: A list of fields that are available to be modified even though they are not mandatory.
+        type: array
+        items:
+          type: string
+      mode:
+        "$ref": "#/definitions/DestinationSyncMode"
+      mandatory_fields:
+        description: A list of fields that need to be provided in order to perform the operation.
+        type: array
+        items:
+          type: string
+      record_matchers:
+        description: If an object can be matched in the destination, this is the list of keys that can be used in order to match the object.
+        type: array
+        items:
```

Comment:
Do we still support nested ids? If this is not the case, I can flatten this and make it support only composite keys as such:
```
      record_matchers:
        description: If an object can be matched in the destination, this is the list of keys that can be used in order to match the object.
        type: array
        items:
          type: array
          items:
            type: string
```

Reply:
Isn't the current code one layer of nesting too much? Seems like two levels of arrays is what we would need: `[ ['key_one', 'level_two'], ['key_two'] ]` would support composite and nested keys, or am I missing something?

Reply:
What you are showing me is how we express one "primary key" currently. I assume we could have multiple "PKs" on which we could match an operation. For example in Salesforce, upserts can be done on any external id fields. In this case, it would look like:
```
[ # root
  [["external_id_1"]], # first external id: this is a non nested, non composite key
  [["external_id_2"]], # second external id: this is a non nested, non composite key  
]
```

Hence, we have three level on nested lists if we support both composite and nested match

Reply:
Got it, I wasn't thinking that there could be _multiple_ ids on which you can match.

Reply:
The user can and need to select only one but there could be many to select from